internal/core/platform.rs

```diff
@@ -287,9 +287,10 @@ pub enum WindowEvent {
         /// # Example
         /// A specific key can be mapped to a unicode by using the [`Key`] enum
         /// ```rust
-        /// let _ = slint::platform::WindowEvent::KeyPressed { text: slint::platform::Key::Shift.into() };
+        /// let _ = slint::platform::WindowEvent::KeyPressed { text: slint::platform::Key::Shift.into(), repeat: false };
         /// ```
         text: SharedString,
+        repeat: bool,
```

Comment:
Sorry for the rookie question, but what do you consider a breaking change, is this struct going to be exported from a dynamic library for some other language interop, or for rust applications directly in some cases?

I can add RepeatedKeyPress, but then isn't it still a breaking change that the old KeyPress won't be used anymore in some cases and that they'll get a new enum (thus breaking auto-repeat)? I'm also not sure what happens for existing binaries that had exhaustive match branches if there is suddenly a new enum type.

I guess you encounter this situation very often, so I guess it's fine. Is it at least necessary to add the enum value at the end to keep the discriminant intact or something? I tried to look up this topic but can't find much using the search terms I'm aware.

Reply:
Done. Since the non-repeated was called `KeyPressed`, I thought `RepeatedKeyPressed` or `KeyPressRepeated` would be better, and went with the latter.

Reply:
This enum is used in the public platform API: https://docs.rs/slint/latest/slint/platform/enum.WindowEvent.html#variant.KeyPressed
It is used by platform author to send events to Slint.
It is a breaking change because, as you probably noted, old code becomes a compilation error.

> I can add RepeatedKeyPress, but then isn't it still a breaking change that the old KeyPress won't be used anymore in some cases and that they'll get a new enum

That would be if there was some kind of event filter.  But right now it is not possible for users to "see" he event, they can only send it to Slint.

> Is it at least necessary to add the enum value at the end to keep the discriminant intact or something?

No.